core/arch/arm/sm/sm_a32.S

```diff
@@ -146,6 +146,25 @@ UNWIND(	.cantunwind)
 	add     r0, sp, #(SM_CTX_NSEC + SM_NSEC_CTX_R8)
 	ldm	r0, {r8-r12}
 
+#ifdef CFG_CORE_WORKAROUND_NSITR_CACHE_PRIME
+	/*
+	 * Prevent leaking information about which code has been executed.
+	 * This is likely to be used together with
+	 * CFG_CORE_WORKAROUND_SPECTRE_BP which also invalidates the branch
+	 * predictor on certain CPUs. In the cases where an alternative
+	 * vector has been installed the branch predictor is already
+	 * invalidated so invalidating here again would be redundant.
+	 */
+#ifdef CFG_CORE_WORKAROUND_SPECTRE_BP
+	read_mvbar r0
+	ldr	r1, =sm_vect_table
```

Comment:
Is it worth it to check if the `bpiall` is needed or not? If so I can rework this a bit more to replace this `ldr` with an `adr` instead.

Reply:
I think that if BP buffers are already flushed when entering monitor with `CFG_CORE_WORKAROUND_SPECTRE_BP=y`. So should be harmless to flush again, hence a simple `write_bpiall` should be to job for all CPUs (even c-a15), conditioned that `CFG_CORE_WORKAROUND_NSITR_CACHE_PRIME=y` forces also `CFG_CORE_WORKAROUND_SPECTRE_BP=y`.

Reply:
OK, that simplifies things. I'll fix.